src/uct/ib/rc/accel/rc_mlx5_ep.c

```diff
@@ -785,11 +793,19 @@ ucs_status_t uct_rc_mlx5_ep_connect_to_ep(uct_ep_h tl_ep,
         return status;
     }
 
-    ep->super.atomic_mr_offset = uct_ib_md_atomic_offset(rc_addr->atomic_mr_id);
+    ep->super.atomic_mr_offset = uct_ib_md_atomic_offset(rc_addr->super.atomic_mr_id);
     ep->super.flags           |= UCT_RC_EP_FLAG_CONNECTED;
-
-    if (uct_ib_md_is_flush_rkey_valid(ep->super.flush_rkey)) {
-        ep->super.flush_rkey |= (uint32_t)rc_addr->atomic_mr_id << 8;
+    addr_length                = UCS_PARAM_VALUE(UCT_EP_CONNECT_TO_EP_PARAM_FIELD,
+                                                 params, ep_addr_length,
+                                                 EP_ADDR_LENGTH,
+                                                 sizeof(uct_rc_mlx5_ep_address_t));
+
+    if ((addr_length > sizeof(uct_rc_mlx5_ep_address_t)) &&
+        (rc_addr->flags & UCT_RC_MLX5_EP_ADDR_FLAG_FLUSH_RKEY)) {
```

Comment:
what is the point of having UCT_RC_MLX5_EP_ADDR_FLAG_FLUSH_RKEY flag, if the field flush_rkey exists anyway?
we should probably use serialize/deserialize rather than struct with all possible fields.

Reply:
this is more flexible for possible extensions, so that UCX takes only that fields which are packed

Reply:
but the field is always present in the struct so tasks up bytes on the wire, why send an "undefined" value?

Reply:
we may want to change it in the future. Maybe define just flag in the struct, and unpack the rest according to the flags

Reply:
this is what i think we should do now - define just the flag in the struct

Reply:
ok, got it in  a wrong way

Reply:
added serialization